crates/primitives/starknet/src/execution/contract_class_wrapper.rs

```diff
@@ -44,104 +47,191 @@ impl<'de> Deserialize<'de> for ProgramWrapper {
+            )),
+            ContractClassWrapper::V1(class) => Self::V1(ContractClassV1(
+                ContractClassV1Inner {
+                    program: class.program.into(),
+                    // Convert EntrypointMapWrapper to HashMap<EntryPointType, Vec<EntryPoint>>
+                    entry_points_by_type: HashMap::from_iter(class.entry_points_by_type.0.iter().clone().map(
+                        |(entrypoint_type, entrypoints)| {
+                            (
+                                entrypoint_type.clone().into(),
+                                entrypoints.clone().into_iter().map(|val| val.into()).collect(),
+                            )
+                        },
+                    )),
+                    hints: class.hints.0.clone(),
+                }
+                .into(),
+            )),
+        }
     }
 }
 
 impl From<ContractClass> for ContractClassWrapper {
     fn from(value: ContractClass) -> Self {
         match value {
-            ContractClass::V0(class) => Self {
+            ContractClass::V0(class) => Self::V0(ContractClassV0Wrapper {
                 program: class.program.clone().into(),
-                entry_points_by_type: EntrypointMapWrapper(unsafe {
+                entry_points_by_type: EntrypointMapV0Wrapper(unsafe {
                     mem::transmute::<
                         HashMap<EntryPointType, Vec<EntryPoint>>,
-                        HashMap<EntryPointTypeWrapper, Vec<EntryPointWrapper>>,
+                        HashMap<EntryPointTypeWrapper, Vec<EntryPointV0Wrapper>>,
                     >(class.entry_points_by_type.clone())
                 }),
-            },
-            _ => Self::default(), // TODO (Greg) handle v1
+            }),
+            ContractClass::V1(class) => Self::V1(ContractClassV1Wrapper {
+                program: class.program.clone().into(),
+                entry_points_by_type: EntrypointMapV1Wrapper(unsafe {
+                    mem::transmute::<
+                        HashMap<EntryPointType, Vec<EntryPointV1>>,
+                        HashMap<EntryPointTypeWrapper, Vec<EntryPointV1Wrapper>>,
+                    >(class.entry_points_by_type.clone())
+                }),
+                hints: class.hints.clone().into(),
+            }),
         }
     }
 }
 
-impl MaxEncodedLen for ContractClassWrapper {
-    fn max_encoded_len() -> usize {
-        // This is the maximum size of a contract in starknet. https://docs.starknet.io/documentation/starknet_versions/limits_and_triggers/
-        Self::MAX_CONTRACT_BYTE_SIZE
+/// Wrapper type for a [HashMap<String, EntryPoint>] object. (It's not really a wrapper it's a
+/// copy of the type but we implement the necessary traits.)
+#[derive(Clone, Debug, PartialEq, Eq, Default, Constructor, Serialize, Deserialize)]
+pub struct EntrypointMapV0Wrapper(pub HashMap<EntryPointTypeWrapper, Vec<EntryPointV0Wrapper>>);
+
+/// SCALE trait.
+impl Encode for EntrypointMapV0Wrapper {
```

Comment:
shouldn't this be in `entrypoint_wrapper.rs` ?

Reply:
Right, the previous wrapper was here as well so I made these here. Actually, all the wrappers will be removed in the next push because greg and lucas are rewriting other repos so we don't need to do this